src/workload.rs

```diff
@@ -78,12 +82,32 @@ impl TryFrom<Option<xds::istio::workload::WorkloadStatus>> for HealthStatus {
     }
 }
 
+#[derive(Debug, Hash, Eq, PartialEq, Clone, Default, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct GatewayAddress {
+    #[serde(default)]
+    pub address: Option<gatewayaddress::Address>,
```

Comment:
Expanding on what I think [John's comment](https://github.com/istio/ztunnel/pull/486/files#r1157494478) was. As far as I can tell a `GatewayAddress` is not really valid if the `address` field is `None`. All the "valid" states for `address` are a `Hostname` or an `IP` which are covered by the inner `enum Address` and the outer `enum Option` really only allows for expressing an invalid state. If there is some case where "None" would be used as a way to represent "localhost" for instance we should just add a more explicit default instead of relying on an implied meaning for `None`.